xc7/yosys/synth.tcl

```diff
@@ -2,21 +2,29 @@ yosys -import
 
 plugin -i xdc
 plugin -i fasm
-#Import the commands from the plugins to the tcl interpreter
-yosys -import
 
-# Map (actually rename) explicitly instantiated IBUFs/OBUFs in order to
-# distinguish them from the inferred ones.
-techmap -map $::env(symbiflow-arch-defs_SOURCE_DIR)/xc7/techmap/iob_map.v
+# Import the commands from the plugins to the tcl interpreter
+yosys -import
 
 # -flatten is used to ensure that the output eblif has only one module.
 # Some of symbiflow expects eblifs with only one module.
-synth_xilinx -vpr -flatten -abc9 -nosrl -noclkbuf -nodsp -iopad
-
-# Map inferred IBUFs/OBUFs to wires. This way ROI targets will work.
-techmap -map $::env(symbiflow-arch-defs_SOURCE_DIR)/xc7/techmap/iob_make_wires.v
-# Map previously renamed IBUFs/OBUFs back to their former names.
-techmap -map $::env(symbiflow-arch-defs_SOURCE_DIR)/xc7/techmap/iob_unmap.v
+#
+# Do not infer IOBs for targets that use a ROI.
+if { $::env(USE_ROI) == "TRUE" } {
+    synth_xilinx -vpr -flatten -abc9 -nosrl -noclkbuf -nodsp
+} else {
+    # Read Yosys baseline library first.
```

Comment:
Why are these not needed for the other pathway?

Reply:
What do you mean by "these"?

Reply:
Line 17+18.